app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingRunnable.java

```diff
@@ -120,16 +129,34 @@ public void run() {
 
             mSyncthing.set(process);
 
-            Thread lInfo = log(process.getInputStream(), Log.INFO, true);
-            Thread lWarn = log(process.getErrorStream(), Log.WARN, true);
+            Thread lInfo = null;
+            Thread lWarn = null;
+            if (returnStdOut) {
+                try {
+                    InputStreamReader isr = new InputStreamReader(process.getInputStream(), Charsets.UTF_8);
+                    BufferedReader br = new BufferedReader(isr);
```

Comment:
Don't you need to close these?

Reply:
Yes. I also added the bufferedReader.close() to the existing code below I didn't touch.
The close also closes the InputStreamReader associated with it (https://docs.oracle.com/javase/8/docs/api/java/io/BufferedReader.html#close--)